src/storage-client/src/controller/command_wals.rs

```diff
@@ -137,9 +128,5 @@ where
         // Remove any shards that are currently in use from shard finalization register.
         self.clear_from_shard_finalization_register(shard_ids_to_keep)
             .await;
-
-        // Determine all shards that are registered that are not in use.
-        self.finalize_shards(registered_shards.difference(&in_use_shards).cloned())
```

Comment:
Why don't we need this anymore? I'm probably just forgetting sth, it has been a while ... 🙈